Building trino-server alone fails: `No tests to run!`

Given

```
# current master
$ git describe
433-103-gbeed389c136

# full clean build
$ ./mvnw clean install -nsu -DskipTests -Dmaven.javadoc.skip=true -Dmaven.source.skip=true -Dair.check.skip-all=true
...
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  05:46 min
[INFO] Finished at: 2023-11-17T16:14:40+01:00
```

now trying to rebuild the trino-server fails

```
$ ./mvnw clean install -pl :trino-server
...
[INFO] --- duplicate-finder:2.0.1:check (default) @ trino-server ---
[INFO] Skipping duplicate-finder execution!
[INFO]
[INFO] --- modernizer:2.7.0:modernizer (modernizer) @ trino-server ---
[INFO]
[INFO] --- surefire:3.2.1:test (default-test) @ trino-server ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.898 s
[INFO] Finished at: 2023-11-17T16:15:34+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.2.1:test (default-test) on project trino-server: No tests to run! -> [Help 1]
[ERROR]
```

--- a/core/trino-server/src/test/java/io/trino/server/TestDummy.java
+++ b/core/trino-server/src/test/java/io/trino/server/TestDummy.java
@@ -1,0 +1,22 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.server;
+
+import org.junit.jupiter.api.Test;
+
+public class TestDummy
+{
+    @Test
+    public void buildRequiresTestToExist() {}
+}

